saleor/tests/e2e/checkout/discounts/test_checkout_calculate_correct_discount_for_sale_and_voucher.py

```diff
@@ -13,17 +17,48 @@
     checkout_delivery_method_update,
     checkout_dummy_payment_create,
     checkout_lines_add,
+    checkout_lines_delete,
+    checkout_lines_update,
 )
 
 
-def prepare_sale_for_variant(
+def prepare_products(
+    e2e_staff_api_client,
+    warehouse_id,
+    channel_id,
+    number_of_products,
```

Comment:
Same here.

Also, it looks like we are using this method multiple times, maybe it should be separated as some helper and reused?